### PR TITLE
Upgrade keycloak

### PR DESCRIPTION
**Pull request by abstractj**
This PR makes the code compatible with the latest changes on Keycloak.

### CODE REVIEW

**Comment by lholmquist** [comment]
For some reason travis is saying it passed,  but i don't think the tests actually ran, looks to be an error with the coveralls step.  looking

**Comment by lholmquist** [comment]
@abstractj when did KC make those changes to the error messages,  in 2.X?  just wondering if we should put some sort of compatibility message about this new version

**Comment by lholmquist** [comment]
@abstractj bro, can you rebase please,

**Comment by abstractj** [comment]
@lholmquist dunno!

**Comment by abstractj** [comment]
@lholmquist that's interesting, even in a fresh environment all the tests pass locally. But as you will notice, they are failing on travis

```

=============================== Coverage summary ===============================
Statements   : 85.47% ( 147/172 )
Branches     : 65.63% ( 42/64 )
Functions    : 96.15% ( 25/26 )
Lines        : 85.47% ( 147/172 )
================================================================================

1..64
# tests 64
# pass  64
```

File: test/clients-test.js [hunk 1]
```diff
@@ -268,6 +267,10 @@ test('Test delete a client', (t) => {
     const id = 'd8c51041-84c7-4e76-901d-401e73eb1666';
 
     client.clients.remove(realmName, id).then(() => {
+    });
```
Comment (lholmquist) [inline test/clients-test.js]: This is going to cause a race condition,  that "client.clients.find" should be inside here

File: test/users-test.js [hunk 2]
```diff
@@ -166,6 +165,11 @@ test('Test delete a user', (t) => {
     const userId = '677e99fd-b854-479f-afa6-74f295052770';
 
     client.users.remove(realmName, userId).then(() => {
+      // Do nothing
```
Comment (lholmquist) [inline test/users-test.js]: same as the above comment, about race conditions

**Comment by abstractj** [comment]
@lholmquist PR updated, but seems like travis is messing with me.

This is the output of running `make ci` locally:

```
=============================== Coverage summary ===============================
Statements   : 86.63% ( 149/172 )
Branches     : 67.19% ( 43/64 )
Functions    : 96.15% ( 25/26 )
Lines        : 86.63% ( 149/172 )
================================================================================

1..66
# tests 66
# pass  66

# ok
```

**Comment by helio-frota** [comment]
@abstractj can you rebase please

**Comment by lance** [comment]
@helio-frota @abstractj there is still a failing test here. Any progress?

**Comment by helio-frota** [comment]
@lance since this was closed https://github.com/bucharest-gold/keycloak-admin-client/issues/13
then would be great to have the bits from @abstractj

**Comment by helio-frota** [comment]
so folks, I've added the @abstractj bits and fixing small changes 
https://github.com/bucharest-gold/keycloak-admin-client/commit/6abafb170b6ed7b9d47895cbb72a6fac05dfa3d8

**Pull request closed without merging**